library/core/src/marker.rs

```diff
@@ -817,6 +817,9 @@ pub trait DiscriminantKind {
 /// This can be used to declare that a constant with a generic type
 /// will not contain interior mutability, and subsequently allow
 /// placing the constant behind references.
+///
+/// This trait is a core part of the language, it just just expressed as a trait in libcore for
+/// convenience. Do *not* implement it for other types.
 #[lang = "freeze"]
 #[unstable(feature = "freeze", issue = "121675")]
```

Comment:
How about some https://github.com/rust-lang/rust/blob/688c30dc9f8434d63bddb65bd6a4d2258d19717c/compiler/rustc_feature/src/builtin_attrs.rs#L828-L835 instead?

Reply:
Well we do have impls just below. What would happen with those?

I think longer-term yes we want `rustc_deny_explicit_impl` but that requires compiler work to turn the existing impls into compiler magic.

Reply:
Oh, right.

Reply:
Maybe put this section under a `# Safety` header to make it more clear?